### ISSUE TITLE
data issue

### ISSUE THREAD

**Issue by meadow163** [atosystem/ThemeTransformer#2]
Hi Ian Shih,
     Thank for your amazing job. 
     I would like to train this model using my data, could you tell my how I make the data like your 'train_seg2_512.pkl' data?
     I will appreciate that if I can your reply.

Thanks.

**Comment by atosystem** [comment]
@meadow163 Thanks for using Theme Transformer!

So I added some codes and information for generating your own training data. (See [preprocess](https://github.com/atosystem/ThemeTransformer/tree/main/preprocess)), or you can see this reply also (https://github.com/atosystem/ThemeTransformer/issues/1#issuecomment-1001791823).

**Comment by chunniunai220ml** [comment]
@atosystem  
Hi, Thank for your share for your novel idea. I would like to know  how to test my data(just convert mp3 to midi)?
and how to get the tempo_json.dict? I did not get it in the POP909 dataset

**Comment by atosystem** [comment]
@chunniunai220ml 

1. Actually, I'm not quite sure what you mean by test our data(just convert mp3 to midi)?
Do you mean that your test themes are in .mp3 format and you want to apply it to Theme Transformer?

2. As for `tempo_json.dict`, it records the main tempo in POP909. I use it as the start tokens for the models. If you are using your own themes, you can assign the tempo manually as you like.

**Comment by chunniunai220ml** [comment]
@atosystem 

Thank for your reply!
question1.  I mean that i have converte the mp3 to the midi format by the webpage :https://www.bearaudiotool.com/zh/mp3-to-midi#google_vignette
and could you tell me what tools you uesed to convert? like your reply in the #issue 1
![1647499274(1)](https://user-images.githubusercontent.com/17668109/158751723-c42c4a62-7b94-47d2-9b6a-00c532793052.jpg)

**Comment by atosystem** [comment]
@chunniunai220ml 
I do not convert mp3 to midi. The midi files I used come from POP909.

For converting mp3 to midi, it's called ***transcription***.
There are a few readily models for piano transcription, you can use ***transcription** as the keyword for searching.

**Comment by atosystem** [comment]
@chunniunai220ml 

So you are asking about converting midi files into the format I showed in the picture (Containing 3 tracks: `MELODY`, `PIANO` and `Theme info track`), right?

The melody and piano tracks come from POP909 directly, as for the theme info track, we used the contrastive learning method for theme retrieval to marked the theme occurrences (as mentioned in the paper).

So, in your case, the midi files are transcribed from your mp3 file. Hence, your might need tools to separate melody from the accompaniment (i.e. the `PIANO` track). A naive way of doing this is to simply consider the notes with the highest pitch on every onset time of the midi file as the melody notes. For more complicated methods, you might need to survey some paper for melody separation or perhaps some DAW software may have this function.
